kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/SchemaUtilsSuite.scala

```diff
@@ -970,14 +990,61 @@ class SchemaUtilsSuite extends AnyFunSuite {
   test("validateUpdatedSchema succeeds when updating field metadata") {
     forAll(validateMetadataChange) { (schemaBefore, schemaAfter) =>
       validateUpdatedSchema(
-        schemaBefore,
-        schemaAfter,
+        metadata(schemaBefore),
+        metadata(schemaAfter),
         emptySet(),
-        emptySet(),
-        metadata(schemaBefore))
+        false /* allowNewNonNullFields */ )
     }
   }
 
+  private val underMaxColIdFieldAdded = Table(
+    ("schemaBefore", "schemaWithUnderMaxFieldIdAdded"),
+    (
+      primitiveSchema,
+      primitiveSchema.add(
+        "required_field",
+        IntegerType.INTEGER,
+        true,
```

Comment:
The validation for new fields being greater than the current max field ID is independent of whether the new field is required or not. I think the test is fine as written (and the logic is right in the actual implementation) but maybe want to mix in a case where the added field is not required.

Reply:
I think I just need to rename this field :) copied it over

But `true` means `nullable` so it's not actually required. I'll update the column names (and also make the fields below nullable I think I missed the others)